### PR TITLE
Scale map to window size

### LINKED ISSUE #110: Map field should be larger then 1x1 pixel

### LINKED ISSUE #112: Map should always be at center of screen
On stratup

### PR DESCRIPTION
**Pull request by kn65op**
Fix #110
Fix #112 

---
This change is [Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115)

### CODE REVIEW

**Review by Abergard — CHANGES_REQUESTED** [review]
I am not convinced to make conversion in that way. I propose to use sf::View here. Here you have my example how it can be used in our project: https://github.com/NadzwyczajnaGrupaRobocza/bomberman/pull/116
I was based on: https://github.com/SFML/SFML/wiki/Source:-Letterbox-effect-using-a-view

> Reviewed 12 of 12 files at r1.
> *[Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-:-LclgRYa-2PVgg3WMee6:b-fwt9ha)* status: all files reviewed, 4 unresolved discussions (waiting on @kn65op)

___
*[src/game/BombermanGameWorld.cpp, line 15 at r1](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-LcjbD3P4XoexP0srqg8:-LcjbD3QC26AQFYLBMoC:b-p38mx2) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/134b4fdc55a17d748843bf7034457ec83450bbc4/src/game/BombermanGameWorld.cpp#L15)):*
> ```cpp
>     std::unique_ptr<graphics::renderer_pool> b)
>     : gen(std::make_unique<BoundaryWallsPositionsGenerator>()),
>       simpleMap{*a, *gen, *b, 100}, ppool{std::move(a)}, rpool{std::move(b)}
> ```

Please name this magic number
___
*[src/game/SimpleMap.hpp, line 18 at r1](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-LcjcDbj6n9FfL-Iubx3:-LcjcDbk7i_FLqYgKA2o:b-koasv5) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/134b4fdc55a17d748843bf7034457ec83450bbc4/src/game/SimpleMap.hpp#L18)):*
> ```cpp
>     SimpleMap(physics::PhysicsEngine& pEngine,
>               const WallPositionsGenerator& wall_positions_generator,
>               graphics::renderer_pool& rEngine, int map_s);
> ```

What about 'init_map_size' variable name?
___
*[src/graphics/core/render_target.stub.hpp, line 36 at r1](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-Lcjd6Hn2OVWjyG-Ofrf:-Lcjd6Ho3IkFs2g7me3s:b-y0qygg) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/134b4fdc55a17d748843bf7034457ec83450bbc4/src/graphics/core/render_target.stub.hpp#L36)):*
> ```
>  std::cout << expectedShape.getPosition() << "\n";
>  std::cout << drawnShape.getPosition() << "\n";
>  std::cout << expectedShape.getSize() << "\n";
> ```

What do you think about using std::clog instead of std::cout here?
___
*[src/graphics/core/sfml_render_target.hpp, line 38 at r1](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-LcjgU89DHVYMIhZSi8W:-LcjgU8B9CaLFE24xaNH:b-oe2r2y) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/134b4fdc55a17d748843bf7034457ec83450bbc4/src/graphics/core/sfml_render_target.hpp#L38)):*
<details><summary><i alt="summary">Quoted 7 lines of code…</i></summary>

> ```
> Type scale(Type val)
>     {
>         auto h_scale = static_cast<float>(size.height) / 100.0f;
>         auto w_scale = static_cast<float>(size.width) / 100.0f;
>         auto sf_vec = static_cast<::sf::Vector2f>(val);
>         return {sf_vec.x * w_scale, sf_vec.y * h_scale};
>     }
> ```
</details>


I am not sure if it works. Windows size may not be a square, but map can be. For example with the 800x600 resolution/window size, the bomberman square tile 1x1 will be scaled to 8x6. Is it what we want to achieve?

**Review by kn65op — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115)* status: 3 of 28 files reviewed, 2 unresolved discussions (waiting on @Abergard)

___
*[src/game/BombermanGameWorld.cpp, line 15 at r1](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-LcjbD3P4XoexP0srqg8:-Lhk3iGY0aPfkzWUJG2B:b-896fix) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/134b4fdc55a17d748843bf7034457ec83450bbc4/src/game/BombermanGameWorld.cpp#L15)):*
<details><summary><i>Previously, Abergard (Lukass) wrote…</i></summary><blockquote>

Please name this magic number
</blockquote></details>

Done.
___
*[src/graphics/core/sfml_render_target.hpp, line 38 at r1](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-LcjgU89DHVYMIhZSi8W:-Lhk3tkL3Y2XxMKugP84:b-ciy9z4) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/134b4fdc55a17d748843bf7034457ec83450bbc4/src/graphics/core/sfml_render_target.hpp#L38)):*
<details><summary><i>Previously, Abergard (Lukass) wrote…</i></summary><blockquote>

> ```
> Type scale(Type val)
>     {
>         auto h_scale = static_cast<float>(size.height) / 100.0f;
>         auto w_scale = static_cast<float>(size.width) / 100.0f;
>         auto sf_vec = static_cast<::sf::Vector2f>(val);
>         return {sf_vec.x * w_scale, sf_vec.y * h_scale};
>     }
> ```

I am not sure if it works. Windows size may not be a square, but map can be. For example with the 800x600 resolution/window size, the bomberman square tile 1x1 will be scaled to 8x6. Is it what we want to achieve?
</blockquote></details>

I can think about algorithm to scale to shorter edge of window in following changes.

File: src/graphics/src/sfml_render_target.hpp [hunk 1]
```diff
@@ -3,17 +3,63 @@
...
+    if (horizontalSpacing)
+    {
+        sizeX = viewRatio / windowRatio;
+        posX = (1 - sizeX) / 2.f;
+    }
+
+    else
+    {
+        sizeY = windowRatio / viewRatio;
+        posY = (1 - sizeY) / 2.f;
```
Comment (moose-x86) [inline src/graphics/src/sfml_render_target.hpp]: it is better to use 0.5F *

File: src/graphics/src/sfml_render_target.hpp [hunk 2]
```diff
@@ -3,17 +3,63 @@
...
+    if (windowRatio < viewRatio)
+        horizontalSpacing = false;
+
+    // If horizontalSpacing is true, the black bars will appear on the left and
+    // right side. Otherwise, the black bars will appear on the top and bottom.
+
+    if (horizontalSpacing)
+    {
+        sizeX = viewRatio / windowRatio;
+        posX = (1 - sizeX) / 2.f;
```
Comment (moose-x86) [inline src/graphics/src/sfml_render_target.hpp]: it is better to use 0.5F *

**Review by Abergard — CHANGES_REQUESTED** [review]
> Reviewed 8 of 10 files at r2, 2 of 3 files at r3, 22 of 22 files at r4, 3 of 3 files at r5.
> *[Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115)* status: all files reviewed, 2 unresolved discussions (waiting on @kn65op)

___
*[src/game/BombermanGameWorld.cpp, line 4 at r4](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-LiMepZR5m3SnV5_s42n:-LiMepZS7nT5mR6AQCCt:b-jdd2dp) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/de9bf3e5c3a77dc3a7506e06466e412fed383696/src/game/BombermanGameWorld.cpp#L4)):*
> ```
> numeric
> ```
Please remove unused includes
___
*[src/game/SimpleMap.cpp, line 20 at r4](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-LiMkcgT6Ay-Cg8W-MIy:-LiMkcgT6Ay-Cg8W-MIz:b-t1w572) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/de9bf3e5c3a77dc3a7506e06466e412fed383696/src/game/SimpleMap.cpp#L20)):*
> ```cpp
> 
>     render_ids.push_back(graphics_engine.acquire(
>         {scale_to_field_size(map_dimensions)}, {0.0f, 0.0f}, map_grey));
> ```

Please use name of Type here. It is not clear what type of parameter are args here.
E.g. ```render_ids.push_back(graphics_engine.acquire(math::Size2f{scale_to_field_size(map_dimensions)}, math::Position2f{0.0f, 0.0f}, map_grey));```

File: src/graphics/src/sfml_render_target.hpp (hunk 1 above)
Comment (kn65op) [reply src/graphics/src/sfml_render_target.hpp]: ~Done.

**Review by kn65op — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115)* status: 26 of 29 files reviewed, 4 unresolved discussions (waiting on @Abergard and @moose-x86)

___
*[src/game/BombermanGameWorld.cpp, line 4 at r4](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-LiMepZR5m3SnV5_s42n:-LiN-abGAEZwgEfTiltN:b-896fix) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/de9bf3e5c3a77dc3a7506e06466e412fed383696/src/game/BombermanGameWorld.cpp#L4)):*
<details><summary><i>Previously, Abergard (Lukass) wrote…</i></summary><blockquote>

> ```
> numeric
> ```
Please remove unused includes
</blockquote></details>

Done.
___
*[src/game/SimpleMap.cpp, line 20 at r4](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-LiMkcgT6Ay-Cg8W-MIy:-LiN-cNx0CimncnW07fL:b-896fix) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/de9bf3e5c3a77dc3a7506e06466e412fed383696/src/game/SimpleMap.cpp#L20)):*
<details><summary><i>Previously, Abergard (Lukass) wrote…</i></summary><blockquote>

Please use name of Type here. It is not clear what type of parameter are args here.
E.g. ```render_ids.push_back(graphics_engine.acquire(math::Size2f{scale_to_field_size(map_dimensions)}, math::Position2f{0.0f, 0.0f}, map_grey));```
</blockquote></details>

Done.
___
*[src/graphics/src/sfml_render_target.hpp, line 43 at r5](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-Lg6k3qLel8d9WFYhRxS-r5-43:-LiN-g3i980XcH_YN_jj:b-896fix) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/dc2e748786769a2bda91f165bf1b41696ac0fb4a/src/graphics/src/sfml_render_target.hpp#L43)):*
<details><summary><i>Previously, moose-x86 (Przemek) wrote…</i></summary><blockquote>

it is better to use 0.5F * 
</blockquote></details>

Done.
___
*[src/graphics/src/sfml_render_target.hpp, line 49 at r5](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-Lg6k3qLel8d9WFYhRxS-r5-49:-LiN-frH3AitiJWA8N7g:b-896fix) ([raw file](https://github.com/nadzwyczajnagruparobocza/bomberman/blob/dc2e748786769a2bda91f165bf1b41696ac0fb4a/src/graphics/src/sfml_render_target.hpp#L49)):*
<details><summary><i>Previously, kn65op (Tomasz Drzewiecki) wrote…</i></summary><blockquote>

~Done.
</blockquote></details>

Done.

**Review by Abergard — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r6, 1 of 1 files at r7.
> *[Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115#-:-LivdjvQ880KCRXqK6Ld:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @moose-x86)

**Review by moose-x86 — APPROVED** [review]

**Review by moose-x86 — APPROVED** [review]

**Review by moose-x86 — APPROVED** [review]
> Reviewed 3 of 12 files at r1, 1 of 10 files at r2, 2 of 3 files at r3, 17 of 22 files at r4, 2 of 3 files at r5, 2 of 3 files at r6, 1 of 1 files at r7, 1 of 1 files at r8.
> *[Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115)* status: :shipit: complete! all files reviewed, all discussions resolved

**Review by Abergard — APPROVED** [review]
> Reviewed 1 of 1 files at r8.
> *[Reviewable](https://reviewable.io/reviews/nadzwyczajnagruparobocza/bomberman/115)* status: :shipit: complete! all files reviewed, all discussions resolved

**Pull request merged**